ledger/eval/eval.go

```diff
@@ -788,13 +788,89 @@ func StartEvaluator(l LedgerForEvaluator, hdr bookkeeping.BlockHeader, evalOpts
 		return nil, fmt.Errorf("overflowed subtracting rewards for block %v", hdr.Round)
 	}
 
+	if eval.eligibleForIncentives(prevHeader.Proposer) {
+		incentive, _ := basics.NewPercent(proto.MiningPercent).DivvyAlgos(prevHeader.FeesCollected)
+		total, o := basics.OAddA(incentive, prevHeader.Bonus)
+		if o {
+			return nil, fmt.Errorf("overflowed adding bonus incentive %d %d", incentive, prevHeader.Bonus)
+		}
+
+		sink, err := eval.state.lookup(prevHeader.FeeSink)
+		if err != nil {
+			return nil, err
+		}
+		available := sink.AvailableBalance(&proto)
+		total = basics.MinA(total, available)
+
+		err = eval.state.Move(prevHeader.FeeSink, prevHeader.Proposer, total, nil, nil)
+		if err != nil {
+			// Should be impossible, we used AvailableBalance
+			return nil, fmt.Errorf("unable to pay block incentive: %v", err)
+		}
+	}
+
+	// Note their proposal if MiningEnabled.
+	if proto.EnableMining {
```

Comment:
Q: if one is not incentive eligible, do we still do all the checks to knock them offline? As of now, looks like we do.

Reply:
Yes, we do. I don't think that's settled law, but I think it's for the best.  The downside is that small accounts will have to deal with challenges, even though they're not even trying to get paid.  I'm empathetic to that, so I'd be ok with only knocking off someone who is `IncentiveEligible` (which also means no existing user, how ignores everything about this stuff and doesn't try for incentives also doesn't need to heartbeat.

Perhaps we can delay this until the heartbeat PR, where it will be clearer what that burden would be.

Reply:
Yeah, lets revisit then. The safer default is in place atm.